Firmware/MotorControl/controller.hpp

```diff
@@ -58,6 +58,7 @@ class Controller : public ODriveIntf::ControllerIntf {
         Controller* parent;
         void set_input_filter_bandwidth(float value) { input_filter_bandwidth = value; parent->update_filter_gains(); }
         void set_steps_per_circular_range(uint32_t value) { steps_per_circular_range = value > 0 ? value : steps_per_circular_range; }
+        void set_control_mode(ControlMode value) { control_mode = value; parent->control_mode_updated(); }
```

Comment:
This sets the input position to the current position estimate every time the control mode is switched. Is this important for some use cases?
I could imagine cases where a user wants to seamlessly transition from velocity control to position control with a preconfigured setpoint.

Reply:
Hmm, seems like a good "Law of Least Surprise" thing anyway.  Switching from vel to pos control could result in the motor suddenly moving, which isn't good for safety.

Reply:
The seamless transition into a standstill position mode, and the safety it provides are both important features for the  Hangprinter use case.

Reply:
Going from not-position control into position control: setting the setpoint to the estimate is fine here. I would say it's better to do this because it's safer, and less surprising like Paul said. The person in your case will have to set the pos setpoint after changing mode, which I think is acceptable.